src/riak_kv_wm_timeseries.erl

```diff
@@ -29,572 +34,375 @@
-                     || {K, V} <- lists:zip(Fields, PEList)],
-                {_, OrderedKeyValues} =
-                    lists:unzip(FVList),
-                {ok, OrderedKeyValues};
-           el/=se ->
-                {error, url_unpaired_keys}
-        end
-    catch
-        throw:ConvertFailed ->
-            {error, ConvertFailed}
-    end.
-
-empair([], Q) -> lists:reverse(Q);
-empair([K, V | T], Q) -> empair(T, [{K, V}|Q]).
-
-convert_fv(Table, Mod, FieldRaw, V) ->
-    Field = [list_to_binary(X) || X <- string:tokens(FieldRaw, ".")],
-    case Mod:is_field_valid(Field) of
-        true ->
-            try
-                convert_field(Table, Field, Mod:get_field_type(Field), V)
-            catch
-                error:badarg ->
-                    %% rethrow with key, for more informative reporting
-                    throw({url_key_bad_value, Table, Field});
-            false ->
-                throw({url_key_bad_key, Table, Field})
-        end
-    end.
-
-convert_field(_T, F, varchar, V) ->
-    {F, list_to_binary(V)};
-convert_field(_T, F, sint64, V) ->
-    {F, list_to_integer(V)};
-convert_field(_T, F, double, V) ->
-    %% list_to_float("42") will fail, so
-    try
-        {F, list_to_float(V)}
+        list_to_float(V)
     catch
         error:badarg ->
-            {F, float(list_to_integer(V))}
+            float(list_to_integer(V))
     end;
-convert_field(T, F, timestamp, V) ->
+convert_field_value(timestamp, V) ->
     case list_to_integer(V) of
-        BadValue when BadValue < 1 ->
-            throw({url_key_bad_value, T, F});
-        GoodValue ->
-            {F, GoodValue}
-    end.
-
-ensure_lk_order_and_strip(LK, FVList) ->
-    [proplists:get_value(F, FVList)
-     || #param_v1{name = F} <- LK].
-
-
--spec valid_params(#wm_reqdata{}, #ctx{}) -> ?CB_RV_SPEC.
-valid_params(RD, Ctx) ->
-    case wrq:get_qs_value("timeout", none, RD) of
-        none ->
-            {true, RD, Ctx};
-        TimeoutStr ->
-            try
-                Timeout = list_to_integer(TimeoutStr),
-                {true, RD, Ctx#ctx{timeout = Timeout}}
-            catch
-                _:_ ->
-                    handle_error({bad_parameter, "timeout"}, RD, Ctx)
-            end
-    end.
-
-
--spec check_permissions(#wm_reqdata{}, #ctx{}) -> {term(), #wm_reqdata{}, #ctx{}}.
-check_permissions(RD, Ctx = #ctx{security = undefined}) ->
-    {true, RD, Ctx};
-check_permissions(RD, Ctx = #ctx{security = Security,
-                                 api_call = Call,
-                                 table = Table}) ->
-    case riak_core_security:check_permission(
-           {riak_kv_ts_util:api_call_to_perm(Call), Table}, Security) of
-        {false, Error, _} ->
-            handle_error(
-              {not_permitted, utf8_to_binary(Error)}, RD, Ctx);
+        GoodValue when GoodValue > 0 ->
+            GoodValue;
         _ ->
-            {true, RD, Ctx}
+            throw(url_key_bad_value)
```

Comment:
Up the stack under the catch in `resource_exists`, you are throwing both a string (in `match_path`), and, here, an atom.